src/granite/querying.cr

```diff
@@ -28,13 +28,14 @@ module Granite::Querying
   # that you are using so you are not restricted or dummied down to support a
   # DSL.
   # Lazy load prevent running unnecessary queries from unused variables.
-  def all(clause = "", params = [] of Granite::Columns::Type)
+  def all(clause = "", params = [] of Granite::Columns::Type, use_primary_adapter = true)
+    switch_to_writer_adapter if use_primary_adapter == true
```

Comment:
```suggestion
    switch_to_writer_adapter if has_performed_recent_write_action
```

Reply:
The idea behind this call (and others like it) is to default to using the primary adapter if the user is calling the function (presumably, if a user wants to write raw sql then defaulting to the writer adapter is a good default). If the user wants to use the reader adapter he has that option.

It is also passed in by other granite methods that call this one. Since most calls to all are from find operations those calls set the variable to false. Which makes granite use whatever the current adapter is. If it happens to be the writer adapter it will use that. But if a write operation has occurred and the time since that operation is sufficient, a callback in granite/base.cr would have switched to the reader adapter.